VideoLocker/src/main/java/org/edx/mobile/module/notification/ParseNotificationDelegate.java

```diff
@@ -133,12 +138,20 @@ public void changeNotificationSetting(String courseId, String channelId, boolean
             pc.setSubscribed(subscribe);
         }
         prefManager.saveNotificationPreference(preference);
-        toggleSubscribeToNotificationServer(pc.getChannelId(), subscribe);
+        changeSubscriptionToNotificationServer(pc.getChannelId(), subscribe, true);
     }
 
+    public void subscribeAndUnsubscribeToServer(String channel, boolean subscribe) {
+        changeSubscriptionToNotificationServer(channel, subscribe, false);
+    }
 
-    public void toggleSubscribeToNotificationServer(final String channel, boolean subscribe) {
-
+    /**
+     * if subscription fails, we may need to update local cache.
+     * @param channel
+     * @param subscribe
+     * @param updateCacheOnError if operation fails, we should update the local cache
```

Comment:
It'd probably be good to have a comment explaining why we want this to be ``false``